### PR TITLE
Internal limits

### PR DESCRIPTION
**Pull request by turtledreams**

### CODE REVIEW

**Comment by turtledreams** [comment]
0 comments 0 changelog etc atm

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
Add missing config options

File: CountlySDKLimitsConfig.m [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        _maxStackTraceLinesPerThread = kCountlyMaxStackTraceLinesPerThread;
+        _maxStackTraceLineLength = kCountlyMaxStackTraceLineLength;
+    }
+    
+    return self;
+}
+
+- (void)setMaxKeyLength:(int)maxKeyLength
+{
+    if (maxKeyLength < 1) {
```
Comment (arifBurakDemiray) [inline CountlySDKLimitsConfig.m:49]: just a question? why not we setting as default if negative number is given? Is not setting it as 1 breaks all things?

File: Countly.m [hunk 2]
```diff
@@ -81,10 +81,22 @@ - (void)startWithConfig:(CountlyConfig *)config
...
+        CountlyCommon.sharedInstance.maxKeyLength = config.maxKeyLength;
+    }
+    if(CountlyCommon.sharedInstance.maxValueLength == kCountlyMaxValueSize && config.maxValueLength != kCountlyMaxValueSize) {
+        CountlyCommon.sharedInstance.maxValueLength = config.maxValueLength;
+    }
+    if(CountlyCommon.sharedInstance.maxSegmentationValues == kCountlyMaxSegmentationValues && config.maxSegmentationValues != kCountlyMaxSegmentationValues) {
+        CountlyCommon.sharedInstance.maxSegmentationValues = config.maxSegmentationValues;
+    }
+    
     CountlyConsentManager.sharedInstance.requiresConsent = config.requiresConsent;
```
Comment (arifBurakDemiray) [inline Countly.m:100]: what about other two limits? Is it accessible through config class?

File: CountlySDKLimitsConfig.m (hunk 1 above)
Comment (ijunaid) [reply CountlySDKLimitsConfig.m:49]: Good catch 👍 
@turtledreams What do you think about it? :)

File: Countly.m (hunk 2 above)
Comment (ijunaid) [reply Countly.m:100]: Which other two limits? :)

File: CountlySDKLimitsConfig.m (hunk 1 above)
Comment (turtledreams) [reply CountlySDKLimitsConfig.m:49]: yeap true seems like I put out some template code here
original check should accept the value if only its greater than 0

File: Countly.m (hunk 2 above)
Comment (arifBurakDemiray) [reply Countly.m:100]: maxStackTraceLinesPerThread and maxStackTraceLineLength

Comment (ijunaid) [reply Countly.m:100]: Yes, we are printing a warning against them in setter that they are just placeholder

**Review by arifBurakDemiray — APPROVED** [review]
What about testing?

**Review by arifBurakDemiray — APPROVED** [review]

**Review by arifBurakDemiray — APPROVED** [review]
I liked the usage of unsigned integer

**Review by peterBrxwn — APPROVED** [review]

**Review by ijunaid — APPROVED** [review]

**Pull request merged**